staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -251,16 +254,89 @@ func (s *Validator) validateExpressions(ctx context.Context, fldPath *field.Path
 			continue
 		}
 		if evalResult != types.True {
-			if len(rule.Message) != 0 {
-				errs = append(errs, field.Invalid(fldPath, sts.Type, rule.Message))
+			if compiled.MessageExpression != nil {
+				messageExpression, newRemainingBudget, msgErr := evalMessageExpression(ctx, compiled.MessageExpression, rule.MessageExpression, activation, remainingBudget)
+				if msgErr != nil {
```

Comment:
The remainingBudget is not handled here?

Reply:
Hmm, you're right. I'll add a test to ensure runtime cost is calculated correctly when a `MessageExpression` evaluation error occurs.

Reply:
(marking as resolved due to https://github.com/kubernetes/kubernetes/pull/115969/commits/f534fc82836446c97210a0faffc906bc9ec81c54)